src/systems/filecoin_vm/interpreter/vm_interpreter.go

```diff
@@ -71,7 +70,7 @@ func (vmi *VMInterpreter_I) ApplyTipSetMessages(inTree st.StateTree, msgs TipSet
 		}
 
 		// Pay block reward.
-		rewardMessage := _makeBlockRewardMessage(outTree, minerOwner, minerPenaltyTotal)
+		rewardMessage := _makeBlockRewardMessage(outTree, blk.Miner(), minerPenaltyTotal)
```

Comment:
must ensure that blk.Miner() in block formation is set as referring to the owner (and not the worker). I believe there are some conflicts with code elsewhere where blk.Miner() is used to get workerKeys() (e.g. in the PoSt verification methods I wrote). Let's agree on a way to resolve that before we merge this in...

Reply:
Right here we're just sending a message to the RewardActor. Responsibility for looking up the owner lies at the receiver. It will need the actor address in order to check for under-collateralisation etc. 

IMO the actor address should be passed ~everywhere, and resolved to an owner or worker last minute.

Reply:
Wait, I might have misunderstood. `blk.Miner` needs to be the miner *actor* address, which is the only unique identifier.

Reply:
yes, ok I agree with that. I expect some cleanup will be required elsewhere...